src/components/main/Main.tsx

```diff
@@ -0,0 +1,108 @@
+import React, { useEffect, useState, useRef } from 'react';
+import { useQuery } from 'react-query';
+
+import { sort } from '@assets/locale/ru.json';
+import { PizzaType, PizzaName, PizzaOrder } from '@ts/types/types';
+import ContentItem from '../contentItem/ContentItem';
+import useRenderButtons from './useRenderButtons';
+import getAllPizzas from '../../core/api';
+import { useAppSelector } from '../../hook';
+import vector from '../../assets/vector.svg';
+
+import { filerPizzas, sortListItems } from '../../config/config';
+
+import './Main.scss';
+
+export default function Main() {
+  const [pizzass, setPizzass] = useState<PizzaType[]>([]);
+  const [isSortListOpen, setIsSortListOpen] = useState(false);
+  const [sortItem, setSortItem] = useState(sortListItems[0]);
+  const sortValue = useRef(null);
+  const order: PizzaOrder[] = useAppSelector((state) => state.pizzas.order);
+  const filter = useAppSelector((state) => state.pizzas.filter);
+  const renderButtons = useRenderButtons(filter);
+  const { data: pizzas } = useQuery<PizzaType[], Error>(
+    'repoData',
+    async () => {
+      try {
+        const pizzasData = await getAllPizzas();
+        setPizzass(pizzasData);
+        return pizzasData;
+      } catch (e) {
+        throw new Error(`Error! status: ${String(e)}`);
+      }
+    }
+  );
+
+  useEffect(() => {
+    const available = filerPizzas[filter];
+    if (filter === 'Все') {
+      setPizzass(pizzas || []);
+    } else {
+      setPizzass(
+        (pizzas || []).filter(({ name }: PizzaName) => {
+          return available?.includes(name);
+        })
+      );
+    }
```

Comment:
здесь стоило бы позаботиться о потоке данных таким образом, чтобы избежать постоянных проверок на существование
```suggestion
    if (filter === 'Все') {
      setPizzass(pizzas);
    } else {
      setPizzass(
        (pizzas).filter(({ name }: PizzaName) => {
          return available?.includes(name);
        })
      );
    }
```

Reply:
![Screenshot_1](https://github.com/AndreyDmitriew/ReactPizza/assets/46495940/2674c49f-37ca-40ca-bd31-66d2e429c522)

this check was created for resolve TS error